sympy/core/power.py

```diff
@@ -564,29 +564,61 @@ def _eval_is_polar(self):
     def _eval_subs(self, old, new):
         from sympy import exp, log, Symbol
         def _check(ct1, ct2, old):
-            """Return bool, pow where, if bool is True, then the exponent of
-            Pow `old` will combine with `pow` so the substitution is valid,
-            otherwise bool will be False,
+            """Return (bool, pow, remainder_pow) where, if bool is True, then the
+            exponent of Pow `old` will combine with `pow` so the substitution
+            is valid, otherwise bool will be False.
+
+            For noncommutative objects, `pow` will be an integer, and a factor
+            `Pow(old.base, remainder_pow)` needs to be included. If there is
+            no such factor, None is returned. For commutative objects,
+            remainder_pow is always None.
 
             cti are the coefficient and terms of an exponent of self or old
             In this _eval_subs routine a change like (b**(2*x)).subs(b**x, y)
             will give y**2 since (b**x)**2 == b**(2*x); if that equality does
             not hold then the substitution should not occur so `bool` will be
             False.
+
             """
             coeff1, terms1 = ct1
             coeff2, terms2 = ct2
             if terms1 == terms2:
-                pow = coeff1/coeff2
-                try:
-                    pow = as_int(pow)
-                    combines = True
-                except ValueError:
-                    combines = Pow._eval_power(
-                        Pow(*old.as_base_exp(), evaluate=False),
-                        pow) is not None
-                return combines, pow
-            return False, None
+                if old.is_commutative:
+                    # Allow fractional powers for commutative objects
+                    pow = coeff1/coeff2
+                    try:
+                        pow = as_int(pow)
+                        combines = True
+                    except ValueError:
+                        combines = Pow._eval_power(
+                            Pow(*old.as_base_exp(), evaluate=False),
+                            pow) is not None
+                    return combines, pow, None
+                else:
+                    # With noncommutative symbols, substitute only integer powers
+                    if not isinstance(terms1, tuple):
```

Comment:
It is not a sequence always, e.g. when called from line 604 with results from `self.exp.as_independent`.